deploy/README.md

```diff
@@ -0,0 +1,221 @@
+url to the API.
+- `clientId` should be provided if you want to enable the 
+[Google Sign-In](https://developers.google.com/identity/sign-in/web/sign-in) feature. This require you to create a 
+valid client ID within the Google cloud project which the Job Manager is deployed to.
+- `dashboardEnabled` controls whether you want to enable the dashboard view in the UI.
+
+**Note:** This file has to be mounted as `/ui/dist/assets/environments/environment.json` in the Job Manager UI docker
+ container during the deployment.
+
+### Capabilities config (required by Cromwell) <a id="capabilities-anchor"></a>
+
+dsub is using hard-coded predefined capabilities configuration for now, so this is only configurable for Cromwell. 
+As the `capabilities-config.json` shows, this file contains the information that UI can fetch from the capabilities
+endpoint of the API, which in turn controls a few behaviors of the UI layer:
+
+- `displayFields` controls which fields would be displayed as columns in the job list view. 
+- `field` is the underlying value of the job's attribute. For Cromwell, use `labels.foo` to refer to a job label `foo`.
+- `display` is the name that will be displayed in the UI.
+- `fieldType`, `validFieldValues`, `editable` and `bulkEditable` are currently only supported by Cromwell backend. 
+Both `editable` and `bulkEditable` will be treated as false unless explicitly set to true, if the field is 
+`editable`, then `fieldType` is required.
+- `commonLabels` controls if the field will become a predefined search key in the query builder.
+- `queryExtensions` represents those ExtendedQueryFields which are queryable, depends on your requirements. Usually the 
+default value should be fine.
+- `authentication` is optional. You should set the `isRequired` to `true` if you want to enable the Google OAuth 
+Sign-In ability.
+
+For Cromwell, the default config looks like:
+```JSON
+{
+  "displayFields": [
+    {
+      "field": "status",
+      "display": "Status"
+    },
+    {
+      "field": "submission",
+      "display": "Submitted"
+    },
+    {
+      "field": "labels.cromwell-workflow-id",
+      "display": "Workflow ID"
+    },
+    {
+      "field": "labels.label",
+      "display": "Label",
+      "editable": true,
+      "bulkEditable": true,
+      "fieldType": "text"
+    },
+    {
+      "field": "labels.flag",
+      "display": "Flag",
+      "editable": true,
+      "bulkEditable": true,
+      "fieldType": "list",
+      "validFieldValues": ["archive", "follow-up"]
+    },
+    {
+      "field": "labels.comment",
+      "display": "Comment",
+      "editable": true,
+      "fieldType": "text"
+    }
+  ],
+  "commonLabels": [
+    "cromwell-workflow-id",
+    "label",
+    "flag"
+  ],
+  "queryExtensions": []
+}
+```
+
+For dsub, depends on the value of the environment variable `PROVIDER_TYPE`, the full version default config looks like:
```

Comment:
Didn't you say earlier that "this isn't relevant for `dsub`?

Reply:
Sorry for the confusion here. Yeah, the `capabilities-config.json` is not relevant for `dsub`, since this part of in dsub is hard-coded in python code: https://github.com/DataBiosphere/job-manager/blob/1f549d20be568aa840e8553669e5bdfdab89c744/servers/dsub/jobs/controllers/capabilities_controller.py#L16 

I thought it would be nice to "translate" the python code as a default `capabilities.json` here so users could be aware of what values they will get as a default. Speaking of this, @bfcrampton I propose we make changes to the dsub controller so it can also accept JSON config files, for both flexibility and consistency reasons.

Reply:
Yup agreed, we should definitely do that for consistency.

Reply:
Cool, going to create an issue for it, and will make a PR in this week.